Add files via upload

SimonBondyUltima opened the pull request:
Updated IA-Connect Microsoft Office for release 9.4


---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [X] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [X] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

SimonBondyUltima committed:
Add files via upload

Updated IA-Connect Microsoft Office for release 9.4

SimonBondyUltima commented:
Hello. I don't understand the failed swagger check. It seems to be unhappy with the new output schema for some of the actions, but we've only changed this (from {} to an object) because we learnt (last time we submitted) that Microsoft now require actions to have at least one output. So, for any actions which we've modified (for example: new inputs), we also changed the output schema to match your new requirements.

vmanoharas commented:
Hello @SimonBondyUltima 

The pipeline was experiencing internal issues, we have resolved them. Could you please resubmit your PR/commit so that it can be triggered again and undergo validation? Thank you for your understanding. We apologize any inconvenience caused.

SimonBondyUltima committed:
Update apiDefinition.swagger.json

Didn't do anything. Just trying to get the checks to re-run

SimonBondyUltima commented:
Hi @vmanoharas 
I re-ran the checks and it failed again. This seems like a valid check to me. I'm changing:
```
        "responses": {
          "200": {
            "description": "Success",
            "schema": {}
          }
```
to instead read:
```
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "type": "object",
              "properties": {
                "MSExcelFormatCellResult": {
                  "type": "boolean",
                  "title": "Command result",
                  "description": "The result of the command (success or failure)."
                },
                "ErrorMessage": {
                  "type": "string",
                  "title": "Error message",
                  "description": "If the command was not successful, this will contain the error message that was returned."
                }
              }
            }
          }
```

i.e. I'm changing a null response to a response which has 2 outputs.
Now I'm only doing this because I was of the understanding that Microsoft would like all actions to have at least one output and hence I'm adding an output.
**If you prefer, I'll only do this for new actions and I'll remove the change to the response for existing actions.**

SimonBondyUltima committed:
Update apiDefinition.swagger.json

Returned the output schema for 200 responses back to the original {} for existing actions (so the schema of existing actions doesn't change)

SimonBondyUltima committed:
Update apiDefinition.swagger.json

Reverted one more output schema back to the way it was set in 9.1. This ensures that the output schema doesn't change between versions

SimonBondyUltima commented:
@vmanoharas I've resolved the remaining issues by putting the output schema back the way it was in release 9.1. This ensures that the automated check which fails when the output schema changes, no longer fails

SimonBondyUltima commented:
Hi. 2 weeks has passed with nothing happening. Is anyone reviewing these PRs? What do I do to get someone's attention?

vmanoharas reviewed (approved):
Hello @SimonBondyUltima, 

I hope you are doing well.
Congratulations, your pull request is approved and merged. Please [submit the files to ISV Studio](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) or [update your submission](https://learn.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#update-submission-prior-to-certification) using your latest commit ID, please allow us up to 1-2 weeks to review your submission, and our Engineers will notify you in the “Activity Control” area in ISV Studio. Please [make sure that the icon meets all the requirements](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector), that it’s size is 230x230, icon background is not white or transparent and it matches the iconBrandColor property in apiProperties.json file.
After making the submission in ISV Studio, please attach an intro.md file in the activity control section. This file is required and it's different from readme.md. You can [find an example of intro.md file on this page](https://docs.microsoft.com/en-us/connectors/custom-connectors/submit-for-certification#submit-to-isv-studio) at step 6.
Please [Create an environment in the Preview region](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#create-an-environment-in-the-preview-region). You will use this environment later to [test your connector](https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-testing#test-your-connector) after Microsoft is done with the functional verification for your connector.
We expect all tests to be completed within 1-2 weeks.
If your connector passes all tests, the deployment process begins, and it typically takes up to 3 to 4 weeks to deploy your connector to all of our regions.
Please let me know if you have any questions or run into any problems while creating or updating your submission.
Thank you very much for working with us.
